dynamic_programming/palindrome_partitioning.cpp

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file
+ * @brief palindrome partitioning algorithm gives you the minimum number of partitions you need to make
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂

Reply:
I have now added a link to geeksforgeeks site that provide proper in-depth algorithm explanation